2020/Day8.py

```diff
@@ -0,0 +1,120 @@
+# https://adventofcode.com/2020/day/8
+import copy
+
+DAY = 8
+
+x_test = '''nop +0
+acc +1
+jmp +4
+acc +3
+jmp -3
+acc -99
+acc +1
+jmp -4
+acc +6'''.split('\n')
+y_test = 5
+
+y_test2 = 8
+
+
+# Part 1
+def parse_lines(INPUT):
+    output = []
+    for i in INPUT:
+        i = i.split()
+        output.append([i[0], int(i[1])])
+
+    return output
+
+
+def execute_program(code):
+    restart = False
+    visited = []
+    i = 0
+    accumulator = 0
+
+    while not restart:
+        action, num = code[i]
+        visited.append(i)
+        if action in ['nop', 'acc']:
+            if action == 'acc':
+                accumulator += num
+            i += 1
+            if (i in visited) or (i == len(code)):
```

Comment:
You repeat this block twice (the `if (i in visited)...`), there is probably a way to clean this up. Also can probably avoid the nested if's by just checking each action independently.

Reply:
Yeah I don't remember why I thought this was necessary. I do know that at first I thought `nop` and `acc` had the same behavior which is why they're grouped together.